rules/build_sector.smk

```diff
@@ -413,6 +436,122 @@ rule build_ates_potentials:
+        "hera_ambient_temperature": f"data/hera_{hera_data_key}/ambient_temp_{hera_data_key}.nc",
+    }
+
+
+rule build_river_heat_potential:
+    params:
+        drop_leap_day=config_provider("enable", "drop_leap_day"),
+        snapshots=config_provider("snapshots"),
+        dh_area_buffer=config_provider("sector", "district_heating", "dh_area_buffer"),
+        generate_temporal_aggregates=config_provider(
+            "plotting", "enable_heat_source_maps"
+        ),
+    input:
+        unpack(input_hera_data),
+        regions_onshore=resources("regions_onshore_base_s_{clusters}.geojson"),
+        dh_areas=resources("dh_areas_base_s_{clusters}.geojson"),
+    output:
+        heat_source_power=resources(
+            "heat_source_power_river_water_base_s_{clusters}.csv"
+        ),
+        heat_source_temperature=resources("temp_river_water_base_s_{clusters}.nc"),
+        heat_source_temperature_temporal_aggregate=resources(
+            "temp_river_water_base_s_{clusters}_temporal_aggregate.nc"
+        ),
+        heat_source_energy_temporal_aggregate=resources(
+            "heat_source_energy_river_water_base_s_{clusters}_temporal_aggregate.nc"
+        ),
+    resources:
+        mem_mb=20000,
+    log:
+        logs("build_river_water_heat_potential_base_s_{clusters}.log"),
+    benchmark:
+        benchmarks("build_river_water_heat_potential_base_s_{clusters}")
+    threads: 2
+    conda:
+        "../envs/environment.yaml"
+    script:
+        "../scripts/build_surface_water_heat_potentials/build_river_water_heat_potential.py"
+
+
+def input_heat_source_temperature(
+    w,
+    replace_names={
+        "air": "air_total",
+        "ground": "soil_total",
+        "ptes": "ptes_top_profiles",
+    },
+):
+
+    heat_pump_sources = set(
+        config_provider("sector", "heat_pump_sources", "urban central")(w)
+    ).union(
+        config_provider("sector", "heat_pump_sources", "urban decentral")(w),
+        config_provider("sector", "heat_pump_sources", "rural")(w),
+    )
+
+    # replace names for soil and air temperature files
+    return {
+        f"temp_{heat_source_name}": resources(
+            "temp_"
+            + replace_names.get(heat_source_name, heat_source_name)
+            + "_base_s_{clusters}"
+            + ("_{planning_horizons}" if heat_source_name == "ptes" else "")
+            + ".nc"
+        )
+        for heat_source_name in heat_pump_sources
+        # remove heat sources with constant temperature - i.e. no temperature profile file (currently only geothermal)
+        if not (
+            heat_source_name
+            in config_provider("sector", "district_heating", "limited_heat_sources")(w)
+            and config_provider(
+                "sector",
+                "district_heating",
+                "limited_heat_sources",
+                heat_source_name,
+                "constant_temperature_celsius",
+            )(w)
+        )
+    }
```

Comment:
might be easier to read (with fewer indentation levels) if moving some bits (like collecting `config_provider` settings out of return statement.